dosing_pump_controller/dosing_pump_controller.ino

```diff
@@ -1,6 +1,71 @@
+//  Created by Andrew Florjancic on 9/17/24.
+//  Copyright © 2024 Andrew Florjancic. All rights reserved
+
+#include <Arduino.h>
+#include <Display.h>
+#include <MenuItem.h>
+#include <Menu.h>
+
+// TODO: Throw all these pins in a struct or something that can represent the Nano Every board that is being used.
+const uint8_t rs = 12, en = 14, d4 = 15, d5 = 16, d6 = 17, d7 = 18;
+const uint8_t lcd_rows = 2, lcd_columns = 16, lcd_backlight = 3; 
+
+LiquidCrystal lcd(rs, en, d4, d5, d6, d7);
+// TODO: Connect contrast pin to one of the Arduino PWM pins - currently controlling LCD contrast with a potentiometer.
+Display display(lcd, lcd_rows, lcd_columns, &lcd_backlight, nullptr);
+
+// Create menus now so that the completion methods can call the menu's present() method
+Menu main_menu("Main Menu", new DoublyLinkedList<MenuItem*>());
+Menu pump1_menu("Pump 1", new DoublyLinkedList<MenuItem*>());
+Menu pump2_menu("Pump 2", new DoublyLinkedList<MenuItem*>());
+Menu pump3_menu("Pump 3", new DoublyLinkedList<MenuItem*>());
+Menu settings_menu("Settings", new DoublyLinkedList<MenuItem*>());
+
+// Returns control to the main_menu, to be used as the completion for menu items added to the main_menu.
+void mainMenuReturnControl() {
```

Comment:
This is going to start adding a lot of clutter to the main program with all the callback functions required. Initially I chose to this route because I was under the impression that whatever subset of C++ Arduino is does not include the standard library. But, it looks like some people have found a way to include parts of the standard library. Also there are some 3rd party libraries that claim to solve this problem but I'd really prefer to keep the number of 3rd party libraries to a minimum. It's not a huge issue that needs to be fixed now so let's explore some options later once we have a working product.